src/db.rs

```diff
@@ -488,7 +501,7 @@ impl DbInner {
                 let mut guard = self.state.write();
                 for kv in wal_replay_buf.iter() {
                     if let Some(ts) = kv.create_ts {
-                        guard.update_clock_tick(ts)?;
+                        last_tick = cmp::max(last_tick, ts);
```

Comment:
Still wondering about this. We are allowing WAL SSTs to have out of order clock ticks? Makes me kinda nervous. I thought we landed on two ticks for txns (read clock on begin, and write clock on commit) to avoid this.

Sorry to keep hammering on this. 🕐 ➡️ 🧠 🤯 .

Reply:
Yeah, we settled on

> I think we should probably raise an SlateDBError::InvalidClockTick. Separate PR sounds good. :)

I'll make that change in the next PR. Or I can just make that as a standalone commit right after this, but it just has a medium API surface area I didn't chase down this PR already is a bit nuanced I didn't want distracting code diffs.

Reply:
Oh! Sorry I misread the part that your comment was referring to. This is when we replay the WAL - here IDK if there's guarantee that the order we flush the wal is the order things are written. I assume it's sorted by key, not by time, so we need to us `cmp::max`.

Reply:
Oh, duh! It's sorted by key 😂 of course. My bad.